app/models/author.rb

```diff
@@ -35,6 +37,20 @@ def leap_year?
     LeapYear.new(created_at).leap_year?
   end
 
+  def zodiac
+    Zodiac.for_date(date_of_birth)
+  end
+
+  def age_at_first_release
+    date_of_first_book = books.map do |book|
+      book.published_at
+    end.min
```

Comment:
ruby has a shorthand for generating blocks that can make things more readable. In this case:

```ruby
date_of_first_book = books.map(&:published_at).min
```